### PR TITLE
Update io.py

### PR DESCRIPTION
**Pull request by freewind2016**
消除重复代码，统一处理逻辑。
1，cvt2frames中直接调用write_frame，避免重复代码，统一逻辑。同时将img为None判断加入write_frame函数。
2，frames2video中直接调用write_frame，避免重复代码，统一逻辑。

### CODE REVIEW

**Comment by dreamerlin** [comment]
please sign in [this](https://cla-assistant.io/open-mmlab/mmcv?pullRequest=847)

File: mmcv/video/io.py [hunk 1]
```diff
@@ -317,7 +314,5 @@ def write_frame(file_idx):
         track_progress(write_frame, range(start, end))
     else:
         for i in range(start, end):
-            filename = osp.join(frame_dir, filename_tmpl.format(i))
-            img = cv2.imread(filename)
-            vwriter.write(img)
+             write_frame(i)
```
Comment (dreamerlin) [inline mmcv/video/io.py:317]: there is a redundant blank

Comment (freewind2016) [reply mmcv/video/io.py]: fixed

**Comment by dreamerlin** [comment]
Seemingly there was another one responsible for this pr @freewind2016

**Comment by freewind2016** [comment]
你好！
我英文不熟，github也没提交过，上面各种提示也搞不懂是什么意思，实在抱歉。

刚开始在win10系统VScode下提交是好的，后来按照文档装了一些YAPF工具等，再提交就报错误，不知怎么解决。所以又切换到Ubuntu虚拟机下重新PR测试，用的账户还是同一个github账号freewind2016。
window下错误信息如下：
————————————
&gt;&nbsp;git&nbsp;show&nbsp;--textconv&nbsp;:mmcv/video/io.py
[INFO]&nbsp;Installing&nbsp;environment&nbsp;for&nbsp;https://github.com/jumanjihouse/pre-commit-hooks.
[INFO]&nbsp;Once&nbsp;installed&nbsp;this&nbsp;environment&nbsp;will&nbsp;be&nbsp;reused.
[INFO]&nbsp;This&nbsp;may&nbsp;take&nbsp;a&nbsp;few&nbsp;minutes...
An&nbsp;unexpected&nbsp;error&nbsp;has&nbsp;occurred:&nbsp;CalledProcessError:&nbsp;command:&nbsp;('rbenv',&nbsp;'init',&nbsp;'-')
return&nbsp;code:&nbsp;1
expected&nbsp;return&nbsp;code:&nbsp;0
stdout:
&nbsp;&nbsp;&nbsp;&nbsp;Executable&nbsp;`rbenv`&nbsp;not&nbsp;found
stderr:&nbsp;(none)
Check&nbsp;the&nbsp;log&nbsp;at&nbsp;C:\Users\Administrator\.cache\pre-commit\pre-commit.log







------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Jintao Lin"<notifications@github.com&gt;; 
发送时间: 2021年2月24日(星期三) 下午4:41
收件人: "open-mmlab/mmcv"<mmcv@noreply.github.com&gt;; 
抄送: "随风"<270118005@qq.com&gt;; "Mention"<mention@noreply.github.com&gt;; 
主题: Re: [open-mmlab/mmcv] Update io.py (#847)





 
Seemingly there was another one responsible for this pr @freewind2016

**Comment by dreamerlin** [comment]
Maybe you can  close this PR and try to reopen a new one same to submit the code

**Comment by dreamerlin** [comment]
I mean close this one and checkout the master to add same changes again for creating a new PR : )

**Comment by OpenMMLab-Coodinator** [comment]
Hi @freewind2016 ！We are grateful for your efforts in helping improve this open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) , where you can share your experiences, ideas, and build connections with like-minded peers. 

To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request closed without merging**